packages/explorer/src/app/(explorer)/[chainName]/worlds/[worldAddress]/observe/useTransactionsWatcher.ts

```diff
@@ -0,0 +1,77 @@
+import { useParams } from "next/navigation";
+import {
+  AbiFunction,
+  Address,
+  BaseError,
+  DecodeFunctionDataReturnType,
+  Hex,
+  Log,
+  Transaction,
+  TransactionReceipt,
+  parseAbiItem,
+} from "viem";
+import { useStore } from "zustand";
+import { useMemo } from "react";
+import { Message } from "../../../../../../observer/messages";
+import { type Write, store as observerStore } from "../../../../../../observer/store";
+import { store as worldStore } from "../store";
+
+export type WatchedTransaction = {
+  writeId: string;
+  hash?: Hex;
+  from?: Address;
+  timestamp?: bigint;
+  transaction?: Transaction;
+  functionData?: DecodeFunctionDataReturnType;
+  value?: bigint;
+  receipt?: TransactionReceipt;
+  status: "pending" | "success" | "reverted" | "rejected" | "unknown";
+  write?: Write;
+  logs?: Log[];
+  error?: BaseError;
+};
+
+export function useTransactionsWatcher() {
+  const { worldAddress } = useParams();
+  const { transactions } = useStore(worldStore);
```

Comment:
afaik this will rerender every time the store changes, not just when `transactions` change

zustand recommends this instead

```suggestion
  const transactions = useStore(worldStore, (state) => state.transactions);
```